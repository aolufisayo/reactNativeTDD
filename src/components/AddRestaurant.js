/**
 * Sample React Native AddRestaurant
 * https://github.com/facebook/react-native
 *
 * @format
 * @flow
 */

import React, { Component } from 'react';
import {
  StyleSheet,
  View,
  Text
} from 'react-native';
import { Button } from 'react-native-elements'
import Reactotron from 'reactotron-react-native'


class AddRestaurant extends Component {
  constructor(props) {
    super(props)

  }

  handlePress = () => {
    const { onButtonPress } = this.props

    onButtonPress()

  }
  render() {
    return (
      <View >
        <Button
          testID="addRestaurantButton"
          title="Add Restaurant"
          onPress={this.handlePress}
        />

      </View>
    );
  }
};

const styles = StyleSheet.create({
  container: {
    justifyContent: 'flex-start',
    alignItems: 'center'
  }
});

export default AddRestaurant;
