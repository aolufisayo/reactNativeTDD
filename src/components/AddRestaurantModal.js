/**
 * Sample React Native AddRestaurant
 * https://github.com/facebook/react-native
 *
 * @format
 * @flow
 */

import React, { Component } from 'react';
import {
  StyleSheet,
  Modal,
  TextInput,
  Text
} from 'react-native';
import { Button, Input } from 'react-native-elements'
import Reactotron from 'reactotron-react-native'


class AddRestaurantModal extends Component {

  constructor(props) {
    super(props)
    this.state = {
      text: ''
    }

  }



  handleSaveText = (text) => {
    this.setState({ text })

  }

  handleSave = () => {
    const { text } = this.state
    const { onSave } = this.props
    Reactotron.log({
      name: 'handleSave from restaurant modal',
      value: this.state.text
    })

    onSave(text)
    this.setState({ text: '' })
  }
  render() {
    const { visible } = this.props
    const { text } = this.state


    return (
      <Modal
        visible={visible}
        animationType='slide'
      >

        <Input
          label="Restaurant Name"
          testID="addRestaurantText"
          value={text}
          onChangeText={this.handleSaveText}
        />
        <Button
          testID="addRestaurantSaveButton"
          title="Save"
          onPress={this.handleSave}
        ></Button>

      </Modal>
    );
  }
};

const styles = StyleSheet.create({
  container: {
    justifyContent: 'flex-start',
    alignItems: 'center'
  }
});

export default AddRestaurantModal;
