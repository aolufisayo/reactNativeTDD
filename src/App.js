/**
 * Sample React Native App
 * https://github.com/facebook/react-native
 *
 * @format
 * @flow
 */

import React, { Component } from 'react';
import {
  StyleSheet,
  View,
  FlatList,
  Button
} from 'react-native';
import Reactotron from 'reactotron-react-native'
import { ListItem } from 'react-native-elements'
import AddRestaurant from './components/AddRestaurant';
import AddRestaurantModal from './components/AddRestaurantModal';


class App extends Component {

  constructor(props) {
    super(props)
    this.state = {
      isVisible: false,
      restaurantNames: []
    }
  }

  onHandlePress = () => {
    this.setState({ isVisible: !this.state.isVisible })

    //console.log(this.state.isVisible)
    Reactotron.log({
      name: 'onHandlePress',
      value: this.state.isVisible
    })
  }

  handleSave = (restaurant) => {
    Reactotron.log({
      name: 'handleSave from app.js',
      value: this.state.restaurantNames
    })
    this.setState((prevState) => (
      {
        isVisible: false,
        restaurantNames: [restaurant, ...prevState.restaurantNames]
      }
    ))
    //console.log("hello", this.props.message)
    Reactotron.log({
      name: 'handleSave from app.js',
      value: restaurant
    })
  }


  render() {
    const { restaurantNames } = this.state
    return (
      <View >
        <AddRestaurant onButtonPress={this.onHandlePress} />

        <AddRestaurantModal visible={this.state.isVisible} onSave={this.handleSave} />


        <FlatList
          data={restaurantNames}
          keyExtractor={(item, index) => index.toString()}
          renderItem={({ item }) => <ListItem title={item} />}
        />

      </View >
    );
  }
};

const styles = StyleSheet.create({
  container: {
    flex: 1
  }
});

export default App;
