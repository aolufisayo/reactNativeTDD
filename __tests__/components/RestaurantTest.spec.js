import React from 'react';
import { shallow } from 'enzyme';
import App from '../../src/App';
import renderer from 'react-test-renderer'
import AddRestaurant from '../../src/components/AddRestaurant';
import AddRestaurantModal from './../../src/components/AddRestaurantModal';

const testID = (id) => {
  return cmp => cmp.props().testID === id
}

describe("Render Restaurant", () => {
  it('should render restaurant component', () => {
    const wrapper = renderer.create(<App />).toJSON()
    expect(wrapper).toMatchSnapshot()

  })


})