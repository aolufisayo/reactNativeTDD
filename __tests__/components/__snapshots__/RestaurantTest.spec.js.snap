// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Render Restaurant should render restaurant component 1`] = `
<View>
  <View>
    <View
      style={
        Object {
          "borderRadius": 3,
        }
      }
    >
      <View
        accessible={true}
        focusable={true}
        isTVSelectable={true}
        onClick={[Function]}
        onResponderGrant={[Function]}
        onResponderMove={[Function]}
        onResponderRelease={[Function]}
        onResponderTerminate={[Function]}
        onResponderTerminationRequest={[Function]}
        onStartShouldSetResponder={[Function]}
        style={
          Object {
            "opacity": 1,
          }
        }
        testID="addRestaurantButton"
      >
        <View
          style={
            Object {
              "alignItems": "center",
              "backgroundColor": "#2089dc",
              "borderColor": "#2089dc",
              "borderRadius": 3,
              "borderWidth": 0,
              "flexDirection": "row",
              "justifyContent": "center",
              "padding": 8,
            }
          }
        >
          <Text
            style={
              Object {
                "color": "white",
                "fontSize": 18,
                "paddingBottom": 1,
                "paddingTop": 2,
                "textAlign": "center",
              }
            }
          >
            Add Restaurant
          </Text>
        </View>
      </View>
    </View>
  </View>
  <RCTScrollView
    data={Array []}
    disableVirtualization={false}
    getItem={[Function]}
    getItemCount={[Function]}
    horizontal={false}
    initialNumToRender={10}
    keyExtractor={[Function]}
    maxToRenderPerBatch={10}
    numColumns={1}
    onContentSizeChange={[Function]}
    onEndReachedThreshold={2}
    onLayout={[Function]}
    onMomentumScrollEnd={[Function]}
    onScroll={[Function]}
    onScrollBeginDrag={[Function]}
    onScrollEndDrag={[Function]}
    removeClippedSubviews={false}
    renderItem={[Function]}
    scrollEventThrottle={50}
    stickyHeaderIndices={Array []}
    updateCellsBatchingPeriod={50}
    viewabilityConfigCallbackPairs={Array []}
    windowSize={21}
  >
    <View />
  </RCTScrollView>
</View>
`;
